core/services/functions/listener.go

```diff
@@ -205,44 +208,37 @@ func (l *FunctionsListener) Start(context.Context) error {
 				},
 				MinIncomingConfirmations: l.pluginConfig.MinIncomingConfirmations,
 			})
+
+			l.shutdownWaitGroup.Add(4)
+			go l.processOracleEventsV0()
+			go l.timeoutRequests()
+			go l.pruneRequests()
+			go func() {
+				<-l.chStop
+				unsubscribeLogs()
+				l.shutdownWaitGroup.Done()
+			}()
+
+			l.mailMon.Monitor(l.mbOracleEvents, "FunctionsListener", "OracleEvents", fmt.Sprint(l.job.ID))
+
+			return nil
+
 		case 1:
-			// TODO: Utilize Log Poller here instead to listen to:
-			// - Router.getContractById(PluginConfig.DONId, false) = current route
-			// - Router.getContractById(PluginConfig.DONId, false) = proposed route
-			// If current = proposed, only listen to current
-			coordinatorContract, err := functions_coordinator.NewFunctionsCoordinator(contractAddress, l.client)
-			if err != nil {
-				return err
-			}
-			unsubscribeLogs = l.logBroadcaster.Register(l, log.ListenerOpts{
-				Contract: coordinatorContract.Address(),
-				ParseLog: coordinatorContract.ParseLog,
-				LogsWithTopics: map[common.Hash][][]log.Topic{
-					functions_coordinator.FunctionsCoordinatorOracleRequest{}.Topic():  {},
-					functions_coordinator.FunctionsCoordinatorOracleResponse{}.Topic(): {},
-				},
-				MinIncomingConfirmations: l.pluginConfig.MinIncomingConfirmations,
-			})
+			l.shutdownWaitGroup.Add(4)
+			go l.processOracleEvents()
+			go l.timeoutRequests()
+			go l.pruneRequests()
+			go func() {
+				<-l.chStop
+				// l.logPollerWrapper.stop() // TODO: do we need teardown in logPollerWrapper?
```

Comment:
TODO comment here:
Do we need to tear down the logPollerWrapper when all of its consumers are no longer using it?

Reply:
no